src/racer.c

```diff
@@ -1195,21 +1798,124 @@ void func_800579B0(unk800579B0 *arg0, UNUSED s32 arg1, f32 arg2) {
+		break;
+    case CAMERA_CAR:
+        // Driving a car.
+        update_camera_car(updateRate, obj, racer);
+        break;
+    case CAMERA_PLANE:
+        // Flying a plane.
+        update_camera_plane(updateRate, obj, racer);
+        break;
+    case CAMERA_FIXED:
+        // Fixes the camera in place, continuing to look at the player. Used when entering doors.
+        update_camera_fixed(updateRate, obj, racer);
+        break;
+    case CAMERA_HOVERCRAFT:
+        // Driving a hovercraft.
+        update_camera_hovercraft(updateRate, obj, racer);
+        break;
+    case CAMERA_FINISH_CHALLENGE:
+        // When you finish a challenge, the camera will rotate around the player, or around a set path, if the player's been KO'd.
+        update_camera_finish_challenge(updateRate, obj, racer);
+        break;
+    case CAMERA_LOOP:
+        // Used for loop-the-loops. Follows the face direction of the player exactly, the standard camera modes do not.
+        update_camera_loop(updateRate, obj, racer);
+        break;
+    case CAMERA_FINISH_RACE:
+        // When you finish a race, the camera will go into a more cinematic mode, following the player.
+        update_camera_finish_race(updateRate, obj, racer);
+        break;
+    }
+    temp_f14 = D_8011D586 / 10240.0f;
+    gCameraObject->x_velocity = (((obj->segment.trans.x_position + (91.75 * racer->ox1) + (90.0 * racer->ox3)) - gCameraObject->trans.x_position) * temp_f14);
+    gCameraObject->z_velocity = (((obj->segment.trans.z_position + (91.75 * racer->oz1) + (90.0 * racer->oz3)) - gCameraObject->trans.z_position) * temp_f14);
+    gCameraObject->y_velocity = (((func_8003ACAC() + 48.5) - gCameraObject->trans.y_position) * temp_f14);
+    gCameraObject->unk38 =  -gCameraObject->trans.x_rotation * temp_f14;
+    gCameraObject->trans.x_position += gCameraObject->x_velocity;
+    gCameraObject->trans.y_position += gCameraObject->y_velocity + gCameraObject->unk30;
+    gCameraObject->trans.z_position += gCameraObject->z_velocity;
+    if (!D_8011D540 && !D_8011D586) {
+        gCameraObject->x_velocity = gCameraObject->x_velocity * 0.95;
+        gCameraObject->y_velocity = gCameraObject->y_velocity * 0.95;
+        gCameraObject->z_velocity = gCameraObject->z_velocity * 0.95;
+    }
+    angle = D_8011D586;
+    if (angle > 0x1400) {
+        angle = 0x2800 - D_8011D586;
+    }
+    if (angle > 0x600) {
+        angle = 0x600;
+    }
+    angle = ((angle >> 4) + 4);
+    delta = updateRate;
+    if (D_8011D583) {
+        D_8011D586 += (delta * angle);
+        if (D_8011D586 > 0x2800) {
+            D_8011D586 = 0x2800;
+        }
+    } else {
+        D_8011D586 -= (delta * angle);
+        if (D_8011D586 < 0) {
+            D_8011D586 = 0;
+        }
+    }
+    gCameraObject->trans.y_rotation -= D_8011D586;
+    gCameraObject->unk3A -= delta;
+    while (gCameraObject->unk3A < 0) {
+        gCameraObject->unk3A += 5;
+         gCameraObject->unk30 = -gCameraObject->unk30 * 0.75;
```

Comment:
This is the most minor of nitpicks ever, but there's an additional space indenting this line.